src/components/MemberProfileEdit.tsx

```diff
@@ -39,193 +78,312 @@ const MemberProfileEdit: React.FC<MemberProfileEditProps> = ({
 
   const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const { name, value } = e.target;
-    setProfile((prevProfile) => ({
-      ...prevProfile,
+    setProfile((prev) => ({
+      ...prev,
       [name]: value,
     }));
   };
 
   const handleSocialFieldChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const [field, key] = e.target.name.split('.');
-
     if (!isSocialField(field as keyof Member)) return;
 
-    setProfile((prevProfile) => ({
-      ...prevProfile,
+    setProfile((prev) => ({
+      ...prev,
       [field]: {
-        ...((prevProfile[field as keyof Member] as SocialField) || {}),
+        ...((prev[field as keyof Member] as SocialField) || {}),
         [key]: e.target.value,
       },
     }));
   };
 
   const handleSocialFieldToggle = (field: keyof Member) => {
     field = field.split('.')[0] as keyof Member;
-
     if (!isSocialField(field)) return;
 
-    setProfile((prevProfile) => {
-      const socialField = prevProfile[field] as SocialField;
-      return {
-        ...prevProfile,
-        [field]: {
-          ...socialField,
-          isPrivate: !socialField.isPrivate,
-        },
-      };
-    });
-  };
-
-  const handleSave = () => {
-    onSave(profile);
+    setProfile((prev) => ({
+      ...prev,
+      [field]: {
+        ...(prev[field] as SocialField),
+        isPrivate: !(prev[field] as SocialField)?.isPrivate,
+      },
+    }));
   };
 
   const githubIsInvalid =
-    profile.github?.username.includes('http') ||
-    profile.github?.username.includes('/');
+    profile.github?.username?.includes('http') ||
```

Comment:
I'm not sure if I like this, what if my github username has http in it. I'd maybe use a helper function like [this regex](https://www.freecodecamp.org/news/check-if-a-javascript-string-is-a-url/) then call it over the string.